src/inventory_ui.h

```diff
@@ -69,35 +78,71 @@ class inventory_entry
             return !( *this == other );
```

Comment:
No, this way it would lead to infinite recursion followed by inevitable stack overflow.